jgrapht-core/src/main/java/org/jgrapht/Graph.java

```diff
@@ -97,9 +98,87 @@
      * defined when the graph is constructed and must not be modified.
      *
      * @return the edge factory using which this graph creates new edges.
+     * @deprecated Use suppliers instead
      */
+    @Deprecated
     EdgeFactory<V, E> getEdgeFactory();
 
+    /**
+     * Return the vertex supplier that the graph uses whenever it needs to create new vertices.
+     * 
+     * <p>
+     * A graph uses the vertex supplier to create new vertex objects whenever a user calls method
+     * {@link Graph#addVertex()}. Users can also create the vertex in user code and then use method
+     * {@link Graph#addVertex(Object)} to add the vertex.
+     * 
+     * <p>
+     * In contrast with the {@link Supplier} interface, the vertex supplier has the additional
+     * requirement that a new and distinct result is returned every time it is invoked. More
+     * specifically for a new vertex to be added in a graph <code>v</code> must <i>not</i> be equal
+     * to any other vertex in the graph. More formally, the graph must not contain any vertex
+     * <code>v2</code> such that <code>v2.equals(v)</code>.
+     * 
+     * @return the vertex supplier or <code>null</code> if the graph has no such supplier
+     */
+    Supplier<V> getVertexSupplier();
+
+    /**
+     * Set the vertex supplier that the graph uses whenever it needs to create new vertices.
+     * 
+     * <p>
+     * A graph uses the vertex supplier to create new vertex objects whenever a user calls method
+     * {@link Graph#addVertex()}. Users can also create the vertex in user code and then use method
+     * {@link Graph#addVertex(Object)} to add the vertex.
+     * 
+     * <p>
+     * In contrast with the {@link Supplier} interface, the vertex supplier has the additional
+     * requirement that a new and distinct result is returned every time it is invoked. More
+     * specifically for a new vertex to be added in a graph <code>v</code> must <i>not</i> be equal
+     * to any other vertex in the graph. More formally, the graph must not contain any vertex
+     * <code>v2</code> such that <code>v2.equals(v)</code>.
+     * 
+     * @param vertexSupplier the vertex supplier
+     */
+    void setVertexSupplier(Supplier<V> vertexSupplier);
```

Comment:
Is there a good reason to expose setVertexSupplier and setEdgeSupplier at the Graph interface level?  It seems like the kind of thing which should be initialized by the graph implementation, rather than called by arbitrary code.  (We never exposed a setEdgeFactory, and no use case ever came up for such a setter.)

Reply:
Removed them, but had to enhance the `GraphDelegator` to be able to override the suppliers.